src/components/TodoItem.tsx

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import classNames from 'classnames';
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo,
+  todos: Todo[],
+  onSetTodos: (newValue: Todo[]) => void
+};
+
+export const TodoItem: React.FC<Props> = (({
+  todo,
+  todos,
+  onSetTodos,
+}) => {
+  const [isEditing, setEditing] = useState(false);
+  const [title, setTitle] = useState('');
+  const [isCompleted, setCompleted] = useState(todo.completed);
+
+  const handleClick = () => {
+    onSetTodos(todos.filter(item => item.id !== todo.id));
+  };
+
+  const handleCheck = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCompleted(event.target.checked);
+  };
```

Comment:
if it is just a toggler, you can simplify this without using an event
```suggestion
  const handleCheck = () => {
    setCompleted(!isCompleted);
  };
```